python311Packages.pydantic: 1.10.12 -> 2.3.0

natsukium opened the pull request:
## Description of changes

- python311Packages.pydantic: 1.10.12 -> 2.3.0
Diff: https://github.com/pydantic/pydantic/compare/refs/tags/v1.10.12...v2.3.0
Changelog: https://github.com/pydantic/pydantic/blob/v2.3.0/HISTORY.md

- python311Packages.versioningit: 2.2.0 -> 2.2.1
Changelog: https://versioningit.readthedocs.io/en/latest/changelog.html

- python311Packages.pydantic-settings: 2.0.3 -> 2.1.0
Diff: https://github.com/pydantic/pydantic-settings/compare/v2.0.3...v2.1.0

- python311Packages.fastapi: add optional-dependencies for pydantic 2

- python311Packages.kanidm: 0.0.3 -> 0.0.3-unstable-2023-08-23

- python311Packages.django-ninja: 0.22.2 -> 1.0.1
Diff: https://github.com/vitalik/django-ninja/compare/v0.22.2...v1.0.1
Changelog: https://github.com/vitalik/django-ninja/releases/tag/v1.0.1

- python311Packages.pythonfinder: 2.0.5 -> 2.0.6
Diff: https://github.com/sarugaku/pythonfinder/compare/refs/tags/2.0.5...2.0.6
Changelog: https://github.com/sarugaku/pythonfinder/blob/v2.0.6/CHANGELOG.rst

- stacs: pin pydantic<2

- python311Packages.pyaussiebb: 0.0.18 -> 0.1.1
Diff: https://github.com/yaleman/pyaussiebb/compare/refs/tags/v0.0.18...v0.1.1
Changelog: https://github.com/yaleman/pyaussiebb/blob/v0.1.1/CHANGELOG.md

- python311Packages.demetriek: apply a pydantic>=2 compatible patch

- python311Packages.aiopurpleair: 2022.12.1 -> 2023.10.0
Diff: https://github.com/bachya/aiopurpleair/compare/refs/tags/2022.12.1...2023.10.0
Changelog: https://github.com/bachya/aiopurpleair/releases/tag/2023.10.0

- python311Packages.aionotion: 2023.05.5 -> 2023.08.0
Diff: https://github.com/bachya/aionotion/compare/2023.05.5...2023.08.0
Changelog: https://github.com/bachya/aionotion/releases/tag/2023.08.0

- python311Packages.fastapi-mail: 1.3.1 -> 1.4.1
Diff: https://github.com/sabuhish/fastapi-mail/compare/refs/tags/1.3.1...1.4.1
Changelog: https://github.com/sabuhish/fastapi-mail/releases/tag/1.4.1

- python311Packages.rstcheck-core: 1.0.3 -> 1.2.0
Diff: https://github.com/rstcheck/rstcheck-core/compare/refs/tags/v1.0.3...v1.2.0
Changelog: https://github.com/rstcheck/rstcheck-core/blob/v1.2.0/CHANGELOG.md

- python311Packages.strawberry-graphql: fix test with pydantic>=2

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

## Eval reports
https://gist.github.com/natsukium/3d7429de0b016cf07492f3561331d002

## Known issue
- [x] cloudflare-dyndns.x86_64-linux
- [x] flare-floss.x86_64-linux
- [x] frigate.x86_64-linux
- [x] home-assistant-component-tests.lametric.x86_64-linux
- [x] home-assistant-component-tests.notion.x86_64-linux
- [x] home-assistant-component-tests.peco.x86_64-linux
- [x] home-assistant-component-tests.zwave_js.x86_64-linux
- [x] khoj.x86_64-linux
- [x] octoprint.x86_64-linux
- [x] python310Packages.qcs-api-client.x86_64-linux
- [x] python310Packages.sqlglot.x86_64-linux
- [ ] python311Packages.bentoml.x86_64-linux
- [ ] python311Packages.correctionlib.x86_64-linux
- [ ] python311Packages.dbt-semantic-interfaces https://github.com/dbt-labs/dbt-semantic-interfaces/issues/134
- [x] python311Packages.deepwave.x86_64-linux
- [ ] python311Packages.dependency-injector.x86_64-linux
- [ ] python311Packages.farm-haystack https://github.com/deepset-ai/haystack/issues/5304
- [ ] python311Packages.flask-security-too.x86_64-linux
- [ ] python311Packages.ha-mqtt-discoverable https://github.com/unixorn/ha-mqtt-discoverable/pull/141
- [ ] python311Packages.huum.x86_64-linux
- [ ] python311Packages.labelbox.x86_64-linux
- [x] python311Packages.linear_operator.x86_64-linux
- [ ] python311Packages.maison.x86_64-linux
- [x] python311Packages.pgmpy.x86_64-linux
- [ ] python311Packages.pycfmodel.x86_64-linux
- [x] python311Packages.pyngo https://github.com/yezz123/pyngo/issues/70
- [ ] python311Packages.pytraccar.x86_64-linux
- [ ] python311Packages.pytradfri https://github.com/home-assistant-libs/pytradfri/issues/720
- [x] python311Packages.radios.x86_64-linux https://github.com/NixOS/nixpkgs/pull/269821
- [ ] python311Packages.shazamio.x86_64-linux
- [ ] python311Packages.steamship https://github.com/steamship-core/python-client/issues/503
- [ ] python311Packages.withings-api.x86_64-linux
- [x] python311Packages.xbox-webapi.x86_64-linux

natsukium committed:
python311Packages.pydantic: 1.10.12 -> 2.3.0

Diff: https://github.com/pydantic/pydantic/compare/refs/tags/v1.10.12...v2.3.0

Changelog: https://github.com/pydantic/pydantic/blob/v2.3.0/HISTORY.md

Drops the documentation build, because it is an ever growing, ever
changing set of dependencies, that we need to package up, hidnering the
upgrade effort.

natsukium committed:
python311Packages.versioningit: 2.2.0 -> 2.2.1

Changelog: https://versioningit.readthedocs.io/en/latest/changelog.html

natsukium committed:
python311Packages.pydantic-settings: 2.0.3 -> 2.1.0

Diff: https://github.com/pydantic/pydantic-settings/compare/v2.0.3...v2.1.0

natsukium committed:
python311Packages.fastapi: add optional-dependencies for pydantic 2

natsukium committed:
python311Packages.kanidm: 0.0.3 -> 0.0.3-unstable-2023-08-23

natsukium committed:
python311Packages.django-ninja: 0.22.2 -> 1.0.1

Diff: https://github.com/vitalik/django-ninja/compare/v0.22.2...v1.0.1

Changelog: https://github.com/vitalik/django-ninja/releases/tag/v1.0.1

natsukium committed:
python311Packages.pythonfinder: 2.0.5 -> 2.0.6

Diff: https://github.com/sarugaku/pythonfinder/compare/refs/tags/2.0.5...2.0.6

Changelog: https://github.com/sarugaku/pythonfinder/blob/v2.0.6/CHANGELOG.rst

natsukium committed:
stacs: pin pydantic_1

natsukium committed:
python311Packages.pyaussiebb: 0.0.18 -> 0.1.1

Diff: https://github.com/yaleman/aussiebb/compare/refs/tags/v0.0.18...v0.1.1

Changelog: https://github.com/yaleman/pyaussiebb/blob/v0.1.1/CHANGELOG.md

SuperSandro2000 commented on pkgs/development/python-modules/pydantic/default.nix:
```diff
@@ -1,124 +1,77 @@
   nativeCheckInputs = [
+    dirty-equals
+    faker
+    pytest-benchmark
```
```suggestion
```
we don't need to waste compute for benchmarking

SuperSandro2000 commented on pkgs/development/python-modules/pydantic/default.nix:
```diff
@@ -1,124 +1,77 @@
 
   pytestFlagsArray = [
-    # https://github.com/pydantic/pydantic/issues/4817
-    "-W" "ignore::pytest.PytestReturnNotNoneWarning"
+    "--benchmark-disable"
   ];
```
I would recommend to rather patch this out to avoid all the extra dependencies of the package

natsukium committed:
python311Packages.demetriek: apply a pydantic>=2 compatible patch

natsukium committed:
python311Packages.demetriek: fix darwin sandbox build

natsukium committed:
python311Packages.aionotion: add meta.changelog

natsukium committed:
python311Packages.fastapi-mail: 1.3.1 -> 1.4.1

Diff: https://github.com/sabuhish/fastapi-mail/compare/refs/tags/1.3.1...1.4.1

Changelog: https://github.com/sabuhish/fastapi-mail/releases/tag/1.4.1

natsukium committed:
python311Pacakges.napari-npe2: 0.7.2 -> 0.7.2-unstable-2023-10-20

Diff: https://github.com/napari/npe2/compare/v0.7.2...9d29e4d6dbbec75c2d36273647efd9ddfb59ded0

natsukium committed:
python311Packages.rstcheck-core: 1.0.3 -> 1.2.0

Diff: https://github.com/rstcheck/rstcheck-core/compare/refs/tags/v1.0.3...v1.2.0

Changelog: https://github.com/rstcheck/rstcheck-core/blob/v1.2.0/CHANGELOG.md

natsukium committed:
python311Packages.strawberry-graphql: fix test with pydantic>=2

natsukium committed:
python311Packages.strawberry-graphql: fix darwin sandbox build

natsukium committed:
remote-exec: pin pydantic_1

augustebaum commented:
Should also help with `python3Packages.pydantic-extra-types` (see [Hydra failure logs](https://hydra.nixos.org/build/240532761/nixlog/1)).

mweinelt commented:
> Should also help with `python3Packages.pydantic-extra-types` (see [Hydra failure logs](https://hydra.nixos.org/build/240532761/nixlog/1)).

Which is only broken because I introduced it before reverting pydantic back to 1.x.

mweinelt commented:
List of newly failing builds between master and this branch.

```
cfripper.x86_64-linux
cloudflare-dyndns.x86_64-linux
dbt.x86_64-linux
flare-floss.x86_64-linux
frigate.x86_64-linux
ha-mqtt-discoverable-cli.x86_64-linux
home-assistant-component-tests.lametric.x86_64-linux
home-assistant-component-tests.notion.x86_64-linux
home-assistant-component-tests.onboarding.x86_64-linux
home-assistant-component-tests.peco.x86_64-linux
home-assistant-component-tests.radio_browser.x86_64-linux
home-assistant-component-tests.traccar.x86_64-linux
home-assistant-component-tests.xbox.x86_64-linux
home-assistant-component-tests.zwave_js.x86_64-linux
khoj.x86_64-linux
ntfy.x86_64-linux
octoprint.x86_64-linux
pgadmin.x86_64-linux
pgadmin4-desktopmode.x86_64-linux
pgadmin4.x86_64-linux
python310Packages.bentoml.x86_64-linux
python310Packages.botorch.x86_64-linux
python310Packages.coffea.x86_64-linux
python310Packages.correctionlib.x86_64-linux
python310Packages.dbt-core.x86_64-linux
python310Packages.dbt-postgres.x86_64-linux
python310Packages.dbt-redshift.x86_64-linux
python310Packages.dbt-semantic-interfaces.x86_64-linux
python310Packages.dbt-snowflake.x86_64-linux
python310Packages.deepwave.x86_64-linux
python310Packages.dependency-injector.x86_64-linux
python310Packages.farm-haystack.x86_64-linux
python310Packages.flask-security-too.x86_64-linux
python310Packages.flyingsquid.x86_64-linux
python310Packages.gpytorch.x86_64-linux
python310Packages.ha-mqtt-discoverable.x86_64-linux
python310Packages.huum.x86_64-linux
python310Packages.labelbox.x86_64-linux
python310Packages.linear_operator.x86_64-linux
python310Packages.maison.x86_64-linux
python310Packages.optuna.x86_64-linux
python310Packages.pgmpy.x86_64-linux
python310Packages.pycfmodel.x86_64-linux
python310Packages.pyngo.x86_64-linux
python310Packages.pytraccar.x86_64-linux
python310Packages.pytradfri.x86_64-linux
python310Packages.qcs-api-client.x86_64-linux
python310Packages.radios.x86_64-linux
python310Packages.shazamio.x86_64-linux
python310Packages.sqlglot.x86_64-linux
python310Packages.steamship.x86_64-linux
python310Packages.withings-api.x86_64-linux
python310Packages.xbox-webapi.x86_64-linux
python310Packages.yamlfix.x86_64-linux
python311Packages.bentoml.x86_64-linux
python311Packages.botorch.x86_64-linux
python311Packages.coffea.x86_64-linux
python311Packages.correctionlib.x86_64-linux
python311Packages.dbt-core.x86_64-linux
python311Packages.dbt-postgres.x86_64-linux
python311Packages.dbt-redshift.x86_64-linux
python311Packages.dbt-semantic-interfaces.x86_64-linux
python311Packages.dbt-snowflake.x86_64-linux
python311Packages.deepwave.x86_64-linux
python311Packages.dependency-injector.x86_64-linux
python311Packages.farm-haystack.x86_64-linux
python311Packages.flask-security-too.x86_64-linux
python311Packages.flyingsquid.x86_64-linux
python311Packages.gpytorch.x86_64-linux
python311Packages.ha-mqtt-discoverable.x86_64-linux
python311Packages.huum.x86_64-linux
python311Packages.labelbox.x86_64-linux
python311Packages.linear_operator.x86_64-linux
python311Packages.maison.x86_64-linux
python311Packages.optuna.x86_64-linux
python311Packages.pgmpy.x86_64-linux
python311Packages.pycfmodel.x86_64-linux
python311Packages.pyngo.x86_64-linux
python311Packages.pytraccar.x86_64-linux
python311Packages.pytradfri.x86_64-linux
python311Packages.radios.x86_64-linux
python311Packages.shazamio.x86_64-linux
python311Packages.steamship.x86_64-linux
python311Packages.withings-api.x86_64-linux
python311Packages.xbox-webapi.x86_64-linux
python311Packages.yamlfix.x86_64-linux
shaq.x86_64-linux
yamlfix.x86_64-linux

mweinelt commented:
Wondering if we want to provide `pydantic_1` for the time being, so we don't accumulate various unmaintained overrides.

natsukium commented:
> Wondering if we want to provide pydantic_1 for the time being, so we don't accumulate various unmaintained overrides.

for application?

dotlambda committed:
python311Packages.xbox-webapi: 2.0.11 -> 2.1.0

Diff: https://github.com/OpenXbox/xbox-webapi-python/compare/v2.0.11...v2.1.0

Changelog: https://github.com/OpenXbox/xbox-webapi-python/blob/v2.1.0/CHANGELOG.md

natsukium committed:
python311Packages.pydantic_1: init at 1.10.3

natsukium committed:
cloudflare-dyndns: pin pydantic_1

natsukium committed:
flare-floss: pin pydantic_1

natsukium committed:
khoj: 0.3.0 -> 1.0.1

Diff: https://github.com/debanjum/khoj/compare/refs/tags/0.3.0...1.0.1

Changelog: https://github.com/debanjum/khoj/releases/tag/1.0.1

natsukium committed:
octoprint: pin pydantic_1

mweinelt commented:
Looked at home-assistant and frigate, and it gets messy fast. I can upgrade frigate to 0.13.0rc6, but for home-assistant I don't think we can fix things without breaking something else.

mweinelt commented:
Downgrade of aionotion and aiopurpleair, as well as fastapi and this is where I stopped.

```patch
commit 222c86ea948b8454314410f76ac05e64ea444e41 (HEAD -> pydantic2)
Author: Martin Weinelt <hexa@darmstadt.ccc.de>
Date:   Tue Nov 28 19:49:23 2023 +0100

    home-assistant: pin pydantic_1
    
    and downgrade
    - aionotion to 2023.05.5
    - aiopurpleair 2022.12.1

diff --git a/pkgs/servers/home-assistant/default.nix b/pkgs/servers/home-assistant/default.nix
index 8f5768fbd2c1..f682cfcf0492 100644
--- a/pkgs/servers/home-assistant/default.nix
+++ b/pkgs/servers/home-assistant/default.nix
@@ -50,6 +50,31 @@ let
         };
       });
 
+      aionotion = super.aionotion.overridePythonAttrs (old: rec {
+        version = "2023.05.5";
+        src = fetchFromGitHub {
+          owner = "bachya";
+          repo = "aionotion";
+          rev = "refs/tags/${version}";
+          hash = "sha256-/2sF8m5R8YXkP89bi5zR3h13r5LrFOl1OsixAcX0D4o=";
+        };
+      });
+
+      aiopurpleair = super.aiopurpleair.overridePythonAttrs (old: rec {
+        version = "2022.12.1";
+        src = fetchFromGitHub {
+          owner = "bachya";
+          repo = "aiopurpleair";
+          rev = "refs/tags/${version}";
+          hash = "sha256-YmJH4brWkTpgzyHwu9UnIWrY5qlDCmMtvF+KxQFXwfk=";
+        };
+        postPatch = ''
+          substituteInPlace pyproject.toml --replace \
+            '"setuptools >= 35.0.2", "wheel >= 0.29.0", "poetry>=0.12"' \
+            '"poetry-core"'
+        '';
+      });
+
       aiowatttime = super.aiowatttime.overridePythonAttrs (oldAttrs: rec {
         version = "0.1.1";
         src = fetchFromGitHub {
@@ -201,6 +226,8 @@ let
         };
       });
 
+      pydantic = super.pydantic_1;
+
       py-synologydsm-api = super.py-synologydsm-api.overridePythonAttrs (oldAttrs: rec {
         version = "2.1.4";
         src = fetchFromGitHub {

natsukium commented:
I tried adding a conditional branch to fastapi for in case of pydantic override, and it might work.
```diff
commit b27a86adf0c7c3b5e726cb2022214a9bf9a5f4d6
Author: natsukium <tomoya.otabi@gmail.com>
Date:   Tue Nov 28 22:33:12 2023 +0900

    fixup! python311Packages.fastapi: add optional-dependencies for pydantic 2

diff --git a/pkgs/development/python-modules/fastapi/default.nix b/pkgs/development/python-modules/fastapi/default.nix
index 3b4d31e59522..f311c9cc92e7 100644
--- a/pkgs/development/python-modules/fastapi/default.nix
+++ b/pkgs/development/python-modules/fastapi/default.nix
@@ -69,6 +69,7 @@ buildPythonPackage rec {
     orjson
     email-validator
     uvicorn
+  ] ++ lib.optionals (lib.versionAtLeast pydantic.version "2") [
     pydantic-settings
     pydantic-extra-types
   ] ++ uvicorn.optional-dependencies.standard;
```

natsukium committed:
python311Packages.aiopurpleair: apply patch for pydantic>=2

natsukium committed:
python311Packages.aionotion: apply patch for pydantic>=2

natsukium committed:
home-assistant: pin pydantic_1

natsukium committed:
frigate: pin pydantic_1

natsukium commented:
@mweinelt Since aionotion and aiopurpleair are the packages whose main consumer is home-assistant, I have reverted the update. I applied a patch for v1 and v2 support instead.
frigate should be able to build now. (Unfortunately, we need to rebuild tensorflow.😟)

mweinelt commented:
Frigate is fine. Will check on home-assistant in a bit.

> Unfortunately, we need to rebuild tensorflow.😟

```
✔ tensorflow-2.13.0 ⏱ 2h42m18s
```
:cry:

mweinelt commented:
Need to pin `versioningit` in home-assistant as well. I can take a look tater today or tomorrow.

```
versioningit> ____________________ ERROR collecting test/test_end2end.py _____________________
versioningit> /nix/store/ghh2dgix6cfhwglxab5aar02y4qa73xb-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py:341: in from_call
versioningit>     result: Optional[TResult] = func()
versioningit> /nix/store/ghh2dgix6cfhwglxab5aar02y4qa73xb-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py:372: in <lambda>
versioningit>     call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
versioningit> /nix/store/ghh2dgix6cfhwglxab5aar02y4qa73xb-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/python.py:531: in collect
versioningit>     self._inject_setup_module_fixture()
versioningit> /nix/store/ghh2dgix6cfhwglxab5aar02y4qa73xb-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/python.py:545: in _inject_setup_module_fixture
versioningit>     self.obj, ("setUpModule", "setup_module")
versioningit> /nix/store/ghh2dgix6cfhwglxab5aar02y4qa73xb-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/python.py:310: in obj
versioningit>     self._obj = obj = self._getobj()
versioningit> /nix/store/ghh2dgix6cfhwglxab5aar02y4qa73xb-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/python.py:528: in _getobj
versioningit>     return self._importtestmodule()
versioningit> /nix/store/ghh2dgix6cfhwglxab5aar02y4qa73xb-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/python.py:617: in _importtestmodule
versioningit>     mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
versioningit> /nix/store/ghh2dgix6cfhwglxab5aar02y4qa73xb-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/pathlib.py:567: in import_path
versioningit>     importlib.import_module(module_name)
versioningit> /nix/store/qp5zys77biz7imbk6yy85q5pdv7qk84j-python3-3.11.6/lib/python3.11/importlib/__init__.py:126: in import_module
versioningit>     return _bootstrap._gcd_import(name[level:], package, level)
versioningit> <frozen importlib._bootstrap>:1204: in _gcd_import
versioningit>     ???
versioningit> <frozen importlib._bootstrap>:1176: in _find_and_load
versioningit>     ???
versioningit> <frozen importlib._bootstrap>:1147: in _find_and_load_unlocked
versioningit>     ???
versioningit> <frozen importlib._bootstrap>:690: in _load_unlocked
versioningit>     ???
versioningit> /nix/store/ghh2dgix6cfhwglxab5aar02y4qa73xb-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/assertion/rewrite.py:186: in exec_module
versioningit>     exec(co, module.__dict__)
versioningit> test/test_end2end.py:95: in <module>
versioningit>     [
versioningit> test/test_end2end.py:95: in <listcomp>
versioningit>     [
versioningit> test/test_end2end.py:78: in mkcases
versioningit>     details = details_cls.model_validate_json(
versioningit> E   AttributeError: type object 'CaseDetails' has no attribute 'model_validate_json'
versioningit> =========================== short test summary info ============================
versioningit> ERROR test/test_end2end.py - AttributeError: type object 'CaseDetails' has no attribute 'model_validate_...
versioningit> !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
versioningit> =============================== 1 error in 0.25s ===============================
```

mweinelt committed:
home-assistant: pin versioningit at 2.2.0

The last version to support pydantic 1.x.

natsukium committed:
treewide: marke as broken due to incompatibility with pydantic>=2

natsukium commented:
@mweinelt It's ready to be merged into python-updates.

mweinelt commented:
Thank you for working that out!

hongquan commented:
You can mark `pyngo` as completed.

SomeoneSerge commented on pkgs/development/python-modules/pydantic/default.nix:
```diff
@@ -1,127 +1,77 @@
-    devtools
-    pyupgrade
+    annotated-types
+    pydantic-core
```
Does it need to be propagated?

mweinelt replied:
When making comments such as this, can you bring forward your research and not make us do our own?

mweinelt replied:
https://github.com/pydantic/pydantic/blob/main/pyproject.toml#L68

SomeoneSerge replied:
Oh I also read this as `poetry-core` last night. Double sorry
